File: crypto/src/main/java/io/oversec/one/crypto/sym/SymmetricKeyPlain.java
package io.oversec.one.crypto.sym;

import io.oversec.one.crypto.symbase.KeyUtil;

import java.io.Serializable;
import java.util.Date;

public class SymmetricKeyPlain implements Serializable {
    private boolean mIsSimpleKey;
    private long id;
    private byte[] raw;

    private Date createdDate;
    private String name;

    public SymmetricKeyPlain() {
    }

    public SymmetricKeyPlain(long id, String name, Date createdDate, byte[] raw, boolean isSimpleKey) {
        this(id, name, createdDate, raw);
        mIsSimpleKey = isSimpleKey;
    }

    public SymmetricKeyPlain(long id, String name, Date createdDate, byte[] raw) {
        this.id = id;
        this.name = name;

        this.createdDate = createdDate;
        this.raw = raw;
    }

    public SymmetricKeyPlain(byte[] raw) {
        this.createdDate = new Date();
        this.raw = raw;
    }


    public Date getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Date createdDate) {
        this.createdDate = createdDate;
    }

    public byte[] getRaw() {
        return raw;
    }

    public void setRaw(byte[] raw) {
        this.raw = raw;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void clearKeyData() {
        KeyUtil.erase(raw);
        raw = null;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public boolean isSimpleKey() {
        return mIsSimpleKey;
    }
}

File: crypto/src/main/java/io/oversec/one/crypto/sym/SymmetricKeyPlain.kt
package io.oversec.one.crypto.sym

import io.oversec.one.crypto.symbase.KeyUtil
import java.io.Serializable
import java.util.Date

class SymmetricKeyPlain : Serializable {
    var isSimpleKey: Boolean = false
    var id: Long = 0
    var raw: ByteArray? = null
    var createdDate: Date? = null
    var name: String? = null

    /*emtpy constructor needed for WaspDb*/
    constructor()

    constructor(
        id: Long,
        name: String,
        createdDate: Date,
        raw: ByteArray,
        isSimpleKey: Boolean
    ) : this(id, name, createdDate, raw) {
        this.isSimpleKey = isSimpleKey
    }

    constructor(id: Long, name: String, createdDate: Date, raw: ByteArray) {
        this.id = id
        this.name = name
        this.createdDate = createdDate
        this.raw = raw
    }

    constructor(raw: ByteArray) {
        this.createdDate = Date()
        this.raw = raw
    }

    fun clearKeyData() {
        KeyUtil.erase(raw)
        raw = null
    }
}
